core/src/main/java/org/infinispan/interceptors/distribution/ScatteredDistributionInterceptor.java

```diff
@@ -714,7 +712,7 @@ private Object handlePutMapOnOrigin(InvocationContext ctx, PutMapCommand command
             localEntries.put(key, new MetadataImmortalCacheValue(entry.getValue(), metadata));
             commitSingleEntryIfNewer(ctxEntry, ctx, command);
          } else if (info.primary() == null) {
-            throw AllOwnersLostException.INSTANCE;
+            throw new AllOwnersLostException(cacheTopology.getTopologyId());
```

Comment:
I guess I'm fixating too much on the "lost" part :)

But IMO we don't need a separate exception every time we need to tell `StateTransferInterceptor` to wait for a new topology, because that just adds duplicate code in `StateTransferInterceptor`. Except here the topology id in `AllOwnersLostException` is the "known bad" topology id and in `OutdatedTopologyException` it's the "expected" topology id, so it's going to be harder to figure out if `STI` is trying to do the same thing for both in the future.